core/bin/qdrant_migrator.rs

```diff
@@ -0,0 +1,434 @@
+                    Some(shadow_write_cluster) => {
+                        let shadow_write_qdrant_client = qdrant_clients
+                            .shadow_write_client(&ds.config().qdrant_config)
+                            .unwrap();
+                        match shadow_write_qdrant_client
+                            .collection_info(ds.qdrant_collection())
+                            .await?
+                            .result
+                        {
+                            Some(info) => {
+                                utils::info(&format!(
+                            "[SHADOW] Qdrant collection: cluster={} collection={} status={}\
+                             points_count={}",
+                            shadow_write_cluster.to_string(),
+                            ds.qdrant_collection(),
+                            info.status.to_string(),
+                            info.points_count,
+                        ));
+                            }
+                            None => Err(anyhow!("Qdrant collection not found"))?,
+                        }
+                    }
+                    None => (),
+                };
+
+                Ok::<(), anyhow::Error>(())
+            }
+            Commands::SetShadowWrite {
+                project_id,
+                data_source_id,
+                cluster,
+            } => {
+                let project = project::Project::new_from_id(project_id);
+                let mut ds = match store.load_data_source(&project, &data_source_id).await? {
+                    Some(ds) => ds,
+                    None => Err(anyhow!("Data source not found"))?,
+                };
+
+                let mut config = ds.config().clone();
+
+                config.qdrant_config = match config.qdrant_config {
+                    Some(c) => Some(QdrantDataSourceConfig {
+                        cluster: c.cluster,
+                        shadow_write_cluster: Some(QdrantCluster::from_str(cluster.as_str())?),
+                    }),
+                    None => Some(QdrantDataSourceConfig {
+                        cluster: QdrantCluster::Main0,
+                        shadow_write_cluster: Some(QdrantCluster::from_str(cluster.as_str())?),
+                    }),
+                };
+
+                // Create collection on shadow_write_cluster.
+                let shadow_write_qdrant_client =
+                    match qdrant_clients.shadow_write_client(&config.qdrant_config) {
+                        Some(client) => client,
+                        None => unreachable!(),
+                    };
+
+                // We send a fake credentials here since this is not really used for OpenAI to get
+                // the embeedding size (which is what happens here). May need to be revisited in
+                // future.
+                let mut credentials = Credentials::new();
+                credentials.insert("OPENAI_API_KEY".to_string(), "foo".to_string());
```

Comment:
this is a little bit of a code smell, but understandable if we don't want to fix

Reply:
It is but it will error if it becomes irrelevant (eg we use another provider that require a credentials to perform this function)